src/CfgMgr32/CfgMgr32+CM_NOTIFY_EVENT_DATA.cs

```diff
@@ -0,0 +1,133 @@
+            public uint Reserved;
+
+            // DeviceInterface
+
+            /// <summary>
+            /// The GUID of the device interface class for the device interface to which the notification event data pertains.
+            /// </summary>
+            [FieldOffset(8)]
+            public Guid ClassGuid;
+
+            /// <summary>
+            /// The first character of the symbolic link path of the device interface to which the notification event data pertains.
+            /// Use <see cref="GetSymbolicLink(CM_NOTIFY_EVENT_DATA*, int)"/> to retrieve this value.
+            /// </summary>
+            [FieldOffset(24)]
+            public char SymbolicLink;
+
+            // DeviceHandle
+
+            /// <summary>
+            /// The GUID for the custom event.
+            /// </summary>
+            [FieldOffset(8)]
+            public Guid EventGuid;
+
+            /// <summary>
+            /// The offset of an optional string buffer. Usage depends on the contract for the EventGuid.
+            /// </summary>
+            [FieldOffset(24)]
+            public int NameOffset;
+
+            /// <summary>
+            /// The number of bytes that can be read from the Data member.
+            /// </summary>
+            [FieldOffset(28)]
+            public int DataSize;
+
+            /// <summary>
+            /// The first bit of optional binary data. Usage depends on the contract for the EventGuid.
+            /// </summary>
+            [FieldOffset(32)]
+            public byte Data;
+
+            // DeviceInstance
+
+            /// <summary>
+            /// The first character of the device instance ID of the device to which the notification event data pertains.
+            /// Use <see cref="GetInstanceId(CM_NOTIFY_EVENT_DATA*, int)"/> to retrieve this value.
+            /// </summary>
+            [FieldOffset(8)]
+            public char InstanceId;
+
+            /// <summary>
+            /// Gets the symbolic link path of the device interface to which the notification event data pertains.
+            /// </summary>
+            /// <param name="eventData">
+            /// The event notification.
+            /// </param>
+            /// <param name="eventDataSize">
+            /// The event notification size.
+            /// </param>
+            /// <returns>
+            /// The symbolic link path of the device interface to which the notification event data pertains
+            /// </returns>
+            public static string GetSymbolicLink(CM_NOTIFY_EVENT_DATA* eventData, int eventDataSize)
+            {
+                if (eventData->FilterType != CM_NOTIFY_FILTER_TYPE.CM_NOTIFY_FILTER_TYPE_DEVICEINTERFACE)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                // Offset and count are represented in characters. Each character is 2 bytes wide.
+                // Trim the terminating \0 character at the end.
```

Comment:
> Trim the terminating \0 character at the end.

Isn't the `\0` character *two* bytes? If so, aren't you trimming off just one of those two bytes with your `- 1` at in the last arg?

Reply:
Yes, it's two bytes wide, but the `string` constructor used here reasons in `char`s, not `byte`s.